File: Practice/09/C++/09.cpp
#include <iostream>
#include <string>
#include <math.h>

using namespace std;
int minutes_from_time(string);

int main()
{   
    string time1, time2;
    cout << "time1: ";
    cin >> time1;
    cout << endl;
    cout << "time2: ";
    cin >> time2;
    cout << endl;

    if (abs(minutes_from_time(time1) - minutes_from_time(time2)) > 15 &&
        abs(minutes_from_time(time1) - minutes_from_time(time2)) < 1425)
    {
        cout << "Meeting will not happen" << endl;
    } else {
        cout << "Meeting will happen" << endl;
    }
}

int minutes_from_time(string time)
{
    size_t sz;

    double hours = stoi(time, &sz);
    double minutes = stoi(time.substr(sz + 1));

    return (hours * 60 + minutes);
}

File: Practice/09/Python/09.py
time1 = input("Время прибытия первого человека: ").split(':')
time2 = input("Время прибытия первого человека: ").split(':')

if int(time1[0]) == 0 and time2[0] == '23':
    time1[0] = '24'
elif int(time2[0]) == 0 and time1[0] == '23':
    time2[0] = '24'

time1 = int(time1[0]) * 60 + int(time1[1])
time2 = int(time2[0]) * 60 + int(time2[1])

if abs(time1 - time2) > 15:
    print("Встреча не состоится")
else:
    print("Встреча состоится")